src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,220 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int numberOfElements;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        numberOfElements = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        if (numberOfElements == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            newNode.previous = tail;
+            tail = newNode;
+        }
+        numberOfElements++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+
```

Comment:
Empty line

Reply:
This blank line uses for create logical grouping, to split variable initialization, and working code.
According to https://mate-academy.github.io/style-guides/java/java.html#s4.6.1-vertical-whitespace